### DISCUSSION TITLE
Oh boy... Not the SLD discussion again!!!!

### DISCUSSION THREAD

**Discussion by vladiuz1** [handshake-org/HIPs#32] [Ideas]
Yes.

SLDs, and third, forth, etc level domains.

My 2 cents in Pre-Pre-Pre HIP. Need to add the softfork considerations, scaling issues that Matt thinks we may bump into.. etc...

Please whoever supports it or completely agains leave your comments. If you have more tech insites and idea please also leave comments so i can edit this gist and evlove it into a HIP that Matt may be able to help finalize if he fixes allt he wallet bugs one day )

Here is the gist up for discussion:

https://gist.github.com/vladiuz1/1f0c23e6cd60e87d74bb68361a1550af

**Comment by Falci** [comment]
First of all: who cares about SLD?

I have `iamfernando` and I'm happy with it. I trust in the server running the NS (mine) and I don't need a decentralised SLD solution.

> Now me, as DNS integrator, would have to integrate both ENS (ethereum based) and Handshake (to make sure the tld is delegated to ENS) into my app.

Eventually, you need to implement the plugins for `.onion`, `.eth` and other reserved name as well.

ENS fork is one possibility, HNS is generic enough to allow you to pick whatever you want.

We could, in theory, start a brand new chain. Focused on SLD only. If there's demand enough, hashpower should appear as well.

**Reply by vladiuz1** [reply]
> First of all: who cares about SLD?

Good question!

You personally don't need SLDs, this is a good point, neither do I. But Handshake is not just about individual use, it was said that DNS servers and TLDs are more for B2B kind of audiences, so businesses do care about SLDs, that's why `foreverdomains.io` was formed, `badass/`, and I guess many others to come. And it's done in such a nice interface, so easy to use, it's just makes total sense.

Now if this trend continues, the audience that could be the users of HNS are going to move to ETH. With XHNS, and further popularisation of Ethereum Handshake may one day port entirely to Ethereum's ENS. If HIP-5 is widely accepted, nobody really needs HNS anymore. Everyone may start pointing their records to a more user-friendly forks of ENS, and have a more usable blockchain with better UX and features. If this will be the case handshake will eventually cease to exist. L2 will no longer need the L1.

If I can point my NS to an ethereum resolver - this ethereum becomes my told NFT eventually that i can pass along, and I can just forget about HNS... This is a nice but self-destructive development for handshake.

**Reply by pinheadmz** [reply]
As I said in the chat but will repeat here: Handshake is a censorship-resistant root zone. Forget ENS for a second, what if we "replaced" the ICANN root zone with a list of domain names and the IP of the nameserver they point to? Oh wait, we just created a root zone file. Even if all 1,500,000 HNS names delegate authority to Ethereum contracts, the HNS blockchain will still be required to store that data verifiably and prove it to light clients, all without trust or central authority that could censor that data. It's nice that for some names like `forever` and `badass` the delegation _probably_ will never change, and those records can essentially be cached forever, but they still need to be served to browsers somehow.

**Comment by mambooos** [comment]
SLD is necessary for domain economy.
